Mathlib/Tactic/Linter/CommandStart.lean

```diff
@@ -0,0 +1,326 @@
+    else
+      parallelScanAux (pushFormatError as (mkFormatError L M "remove line break")) ls.trimLeft M
+  | l, m => -- `l` is not whitespace
+    if l == m then
+      parallelScanAux as ls ms
+    else
+      if m.isWhitespace then
+        parallelScanAux (pushFormatError as (mkFormatError L M "missing space")) L ms.trimLeft
+    else
+      pushFormatError as (mkFormatError ls ms "Oh no! (Unreachable?)")
+
+@[inherit_doc parallelScanAux]
+def parallelScan (src fmt : String) : Array FormatError :=
+  parallelScanAux ∅ src fmt
+
+namespace Style.CommandStart
+
+/--
+`unlintedNodes` contains the `SyntaxNodeKind`s for which there is no clear formatting preference:
+if they appear in surface syntax, the linter will ignore formatting.
+
+Currently, the unlined nodes are mostly related to `Subtype`, `Set` and `Finset` notation and
+list notation.
+-/
+abbrev unlintedNodes := #[
+  -- # set-like notations, have extra spaces around the braces `{` `}`
+
+  -- subtype, the pretty-printer prefers `{ a // b }`
+  ``«term{_:_//_}»,
+  -- set notation, the pretty-printer prefers `{ a | b }`
+  `«term{_}»,
+  -- empty set, the pretty-printer prefers `{ }`
+  ``«term{}»,
+  -- set builder notation, the pretty-printer prefers `{ a : X | p a }`
+  `Mathlib.Meta.setBuilder,
+
+  -- # misc exceptions
+
+  -- We ignore literal strings.
+  `str,
+
+  -- list notation, the pretty-printer prefers `a :: b`
+  ``«term_::_»,
+
+  -- negation, the pretty-printer prefers `¬a`
+  ``«term¬_»,
+
+  -- declaration name, avoids dealing with guillemets pairs `«»`
+  ``Parser.Command.declId,
+
+  `Mathlib.Tactic.superscriptTerm, `Mathlib.Tactic.subscript,
+
+  -- notation for `Bundle.TotalSpace.proj`, the total space of a bundle
+  -- the pretty-printer prefers `π FE` over `π F E` (which we want)
+  `Bundle.termπ__,
+
+  -- notation for `Finset.slice`, the pretty-printer prefers `𝒜 #r` over `𝒜 # r` (mathlib style)
+  `Finset.«term_#_»,
+
+  -- The docString linter already takes care of formatting doc-strings.
+  ``Parser.Command.docComment,
+
+  -- `omit [A] [B]` prints as `omit [A][B]`, see https://github.com/leanprover/lean4/pull/8169
+  ``Parser.Command.omit,
+  ]
+
+/--
+Given an array `a` of `SyntaxNodeKind`s, we accumulate the ranges of the syntax nodes of the
+input syntax whose kind is in `a`.
+
+The linter uses this information to avoid emitting a warning for nodes with kind contained in
+`unlintedNodes`.
+-/
+def getUnlintedRanges(a : Array SyntaxNodeKind) :
+    Std.HashSet String.Range → Syntax → Std.HashSet String.Range
+  | curr, s@(.node _ kind args) =>
+    let new := args.foldl (init := curr) (·.union <| getUnlintedRanges a curr ·)
+    if a.contains kind then
+      new.insert (s.getRange?.getD default)
+    else
+      new
+  | curr, _ => curr
+
+/-- Given a `HashSet` of `String.Ranges` `rgs` and a further `String.Range` `rg`,
+`outside rgs rg` returns `true` if and only if `rgs` contains a range the completely contains
```

Comment:
Let's make the name and documentation match:
```suggestion
`outside? rgs rg` returns `true` if and only if `rgs` contains a range the completely contains
```
That said, I'd expect a name `isOutside` (as `outside?` makes me think of an `Option` return value.)
Taking a step back, how about a name like `isSubsumedBy rg rgs` (note the argument order change)?

Reply:
Actually: `isOutside` also works, when you flip the meaning of the return value in the right way.

Right now, documentation and code say different things. Either way is fine, but they should be consistent :-)